### ISSUE TITLE
Produce pretty-printed Debug output

### ISSUE THREAD

**Issue by rubenrua** [mongodb/bson-rust#106]
Like `serde_json`, add support for:

```rust
format!("{:#}", bson);
```

```
[
    Object(
        {
            "contributors": Array(
                [
                    Object(
                        {
                            "date": String(
                                "2012-03-28"
                            ),
                            "role": String(
                                "author"
                            )
                        }
                    ),
...
```

More info about pretty-printed debug: https://www.reddit.com/r/rust/comments/3ceaui/psa_produces_prettyprinted_debug_output/

`serde_json` [implementation](https://github.com/serde-rs/json/blob/master/src/value/mod.rs#L275) and [issue](https://github.com/serde-rs/json/issues/279).

**Comment by saghm** [comment]
I've filed https://jira.mongodb.org/browse/RUST-282 to track the work for this issue.

**Comment by patrickfreed** [comment]
This was included in `2.0.0-beta.2`.

e.g.
```rust
let bson = bson!({                 
    "x": 1,                        
    "array": [true, 5.5, "ok"],    
    "doc": {                       
        "a": true,                 
        "b": bson::DateTime::now(),
    }                              
});                                
println!("{:#?}", bson);           
```
prints
```
Document({
    "x": Int32(
        1,
    ),
    "array": Array([
        Boolean(
            true,
        ),
        Double(
            5.5,
        ),
        String(
            "ok",
        ),
    ]),
    "doc": Document({
        "a": Boolean(
            true,
        ),
        "b": DateTime(
            2021-08-06T19:16:37.551Z,
        ),
    }),
})
```
